tests/functional/test_report_only_no_errors.py

```diff
@@ -0,0 +1,319 @@
+    bot_token = os.environ.get("BOT_TOKEN")
+    if not bot_token:
+        bot_name = "github-actions[bot]"
+        bot_token = os.environ.get("GITHUB_TOKEN")
+        if not bot_token:
+            raise Exception("BOT_TOKEN environment variable not defined")
+
+    pr_base_branch = 'test-charts'
+    fork_branch = pr_base_branch + '-pr'
+    test_repo = str(base64.b64decode(test_repo), encoding="utf-8")
+    fork_repo = str(base64.b64decode(fork_repo), encoding="utf-8")
+    yield Secret(test_repo, fork_repo, cluster_token, bot_token, bot_name, pr_base_branch, fork_branch)
+
+    # Teardown step to cleanup branches
+    headers = {'Accept': 'application/vnd.github.v3+json',
+               'Authorization': f'Bearer {bot_token}'}
+    requests.delete(
+        f'https://api.github.com/repos/{test_repo}/git/refs/heads/{pr_base_branch}-gh-pages', headers=headers)
+    requests.delete(
+        f'https://api.github.com/repos/{fork_repo}/git/refs/heads/{fork_branch}', headers=headers)
+
+
+@scenario('features/report_without_chart.feature', "The partner hashicorp submits a error-free report for the vault chart")
+def test_partner_submits_report_without_any_errors():
+    """The partner hashicorp submits a error-free report for the vault chart."""
+
+
+@scenario('features/report_without_chart.feature', "The redhat associate submits a error-free report for the vault chart")
+def test_redhat_submits_report_without_any_errors():
+    """The redhat associate submits a error-free report for the vault chart."""
+
+
+@given("hashicorp is a valid partner")
+def hashicorp_is_a_valid_partner(secrets):
+    secrets.vendor_type = 'partners'
+    secrets.vendor = 'hashicorp'
+
+
+@given("a redhat associate has a valid identity")
+def hashicorp_is_a_valid_partner(secrets):
+    secrets.vendor_type = 'redhat'
+    secrets.vendor = 'redhat'
+
+
+@given("hashicorp has created a error-free report")
+@given("the redhat associate has created a report without any errors")
+def the_user_has_created_a_report_without_errors(secrets):
+    """The user has created a report without any errors."""
+
+    repo = git.Repo(os.getcwd())
+    if os.environ.get('WORKFLOW_DEVELOPMENT'):
+        logger.info("Wokflow development enabled")
+        repo.git.add(A=True)
+        repo.git.commit('-m', 'Checkpoint')
+
+    headers = {'Accept': 'application/vnd.github.v3+json',
+               'Authorization': f'Bearer {secrets.bot_token}'}
+
+    # Get SHA from 'dev-gh-pages' branch
+    logger.info(
+        f"Create '{secrets.test_repo}:{secrets.pr_base_branch}-gh-pages' from '{secrets.test_repo}:dev-gh-pages'")
+    r = requests.get(
+        f'https://api.github.com/repos/{secrets.test_repo}/git/ref/heads/dev-gh-pages', headers=headers)
+    j = json.loads(r.text)
+    sha = j['object']['sha']
+
+    # Create a new gh-pages branch for testing
+    data = {'ref': f'refs/heads/{secrets.pr_base_branch}-gh-pages', 'sha': sha}
+    r = requests.post(
```

Comment:
Can we abstract away from the test cases the github functionality that we need to use over and over on all the test cases?

Reply:
sounds good to me, will add a helper function to wrap around the github api calls